test/test_roscpp/test/src/multiple_latched_publishers.cpp

```diff
@@ -83,10 +51,17 @@ TEST(MultipleLatchedPublishers, LatchedPublisherReceiveMultiple)
     nh.advertise<bool>("foo", 10, true),
     nh.advertise<bool>("foo", 10, true),
   };
+  std::vector<ros::Publisher> unlatched_publishers = {
+    nh.advertise<bool>("foo", 10, false),
+    nh.advertise<bool>("foo", 10, false),
+  };
 
   for (auto& pub : latched_publishers) {
     pub.publish(true);
   }
+  for (auto& pub : unlatched_publishers) {
+    pub.publish(true);
+  }
```

Comment:
The updated test case does not verify whether the callback for the late subscribers is called only for the `latched_publishers`. It is still possible that it would have been called a 4th or 5th time after the `received_promise` is set for the `unlatched_publishers`.

The previous version also did not explicitly cover this. But there have been only three publishers, three publish calls and three callbacks were expected. Now there are five publishers and five publish calls but still only three expected callbacks (and not more).

Reply:
I'm going to look into this, but I don't think it's actually possible to get latching behaviour from the non-latching publishers, since they check whether they are latched before even saving the `last_message_`:

https://github.com/ros/ros_comm/blob/3fc96bbe7870a5d948d45d00b0e15103193fb4cc/clients/roscpp/src/libros/publisher.cpp#L103-L106

So even if `pushLastMessage` is mistakenly called (which also shouldn't be able to happen), there'd be no way to get more than the expected number of latched responses.

That said, we could definitely update the test to wait around and confirm that no additional responses arrive.